File: web/nexus-ui/src/contracts/requests/recovery-password.request.ts
export interface RecoveryPasswordRequest{
    login: string;
    // Srp
    encryptedVerifier: string;
    srpSalt: string;
    srpVersion: number,
    encryptedVerifierWrapKey: string, 
    keyWrapVersion: number, 
    asymmetricKeyId: string,
    // Dek 
    encryptedDek: string;
    dekSalt: string;
    cryptoVersion: number;
    // RecoveryKeys
    recoveryKeys: Array<{encryptedValue: string, cryptoVersion: number}>;
}

File: backend/src/shared/contracts/Shared.Contracts/RecoveryPasswordRequest.cs
namespace Shared.Contracts
{
    public sealed record RecoveryPasswordRequest(
        string Login, 
        string Verifier, 
        string ClientSalt, 
        string EncryptedDek, 
        int CryptoVersion, 
        int SrpVersion, 
        string EncryptedVerifierWrapKey, 
        int KeyWrapVersion, 
        string AsymmetricKeyId,
        IReadOnlyCollection<RecoveryKeysRequestData> RecoveryKeys);

    public sealed record RecoveryKeysRequestData(string EncryptedValue, int CryptoVersion); 
}